instance/tests/models/test_server.py

```diff
@@ -306,38 +306,46 @@ def test_reboot_server_wrong_status(self, server_status):
         with self.assertRaises(WrongStateException):
             server.reboot()
 
-    def test_terminate_pending_server(self):
+    @data(
+        ('booting-server-id', ServerStatus.Booting),
+        ('ready-server-id', ServerStatus.Ready),
+    )
+    @unpack
+    def test_terminate_server_vm_available(self, openstack_id, server_status):
         """
-        Terminate a server with a 'pending' status
+        Terminate a server that was created successfully
         """
-        server = OpenStackServerFactory()
+        server = OpenStackServerFactory(openstack_id=openstack_id, status=server_status)
         server.terminate()
         self.assertEqual(server.status, ServerStatus.Terminated)
-        server.terminate() # This shouldn't change anything
-        self.assertEqual(server.status, ServerStatus.Terminated)
-        self.assertFalse(server.nova.mock_calls)
+        server.os_server.delete.assert_called_once_with()
 
     @data(
+        ('pending-server-id', ServerStatus.Pending),
         ('building-server-id', ServerStatus.Building),
-        ('booting-server-id', ServerStatus.Booting),
         ('failed-server-id', ServerStatus.BuildFailed),
-        ('ready-server-id', ServerStatus.Ready),
+        ('terminated-server-id', ServerStatus.Terminated),
     )
     @unpack
-    def test_terminate_server(self, openstack_id, server_status):
+    def test_terminate_server_vm_unavailable(self, openstack_id, server_status):
         """
-        Terminate a server
+        Terminate a server that is unavailable
         """
         server = OpenStackServerFactory(openstack_id=openstack_id, status=server_status)
         server.terminate()
         self.assertEqual(server.status, ServerStatus.Terminated)
-        server.os_server.delete.assert_called_once_with()
+        server.os_server.delete.assert_not_called()
```

Comment:
This seems mostly right, but if we terminate a server that is `Building`, shouldn't `os_server.delete` be called in that case? Otherwise, the server will finish building and then we'll have an orphaned VM in OpenStack that never gets deleted. Or am I missing something?